### PR TITLE
Added annotation subkind for Liberty

### PR DESCRIPTION
**Pull request by JusteenR**

### CODE REVIEW

File: pkg/controller/openliberty/libertyapplication_controller.go [hunk 1]
```diff
@@ -456,5 +456,35 @@ func (r *ReconcileOpenLiberty) Reconcile(request reconcile.Request) (reconcile.R
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", routev1.SchemeGroupVersion.String()))
 	}
 
+	if instance.Spec.CreateAppDefinition != nil && *instance.Spec.CreateAppDefinition {
+		m := make(map[string]string)
+		m["kappnav.subkind"] = "Liberty"
+		if instance.Spec.Storage != nil {
+			statefulSet := &appsv1.StatefulSet{ObjectMeta: defaultMeta}
```
Comment (arturdzm) [inline pkg/controller/openliberty/libertyapplication_controller.go]: You can add this logic into existing place where we already define statefulset/deployment

Comment (JusteenR) [reply pkg/controller/openliberty/libertyapplication_controller.go]: Added the changes into where statefulset and deployment are defined.

File: pkg/controller/openliberty/libertyapplication_controller.go [hunk 2]
```diff
@@ -346,6 +346,11 @@ func (r *ReconcileOpenLiberty) Reconcile(request reconcile.Request) (reconcile.R
 			autils.CustomizeStatefulSet(statefulSet, instance)
 			autils.CustomizePodSpec(&statefulSet.Spec.Template, instance)
 			autils.CustomizePersistence(statefulSet, instance)
+			if instance.Spec.CreateAppDefinition != nil && *instance.Spec.CreateAppDefinition {
```
Comment (leochr) [inline pkg/controller/openliberty/libertyapplication_controller.go]: @JusteenR Users don't have to set `CreateAppDefinition` explicitly. It should be assumed to be true by default. Please check if it's nil or true. Thanks.

Comment (JusteenR) [reply pkg/controller/openliberty/libertyapplication_controller.go]: Changed to account for the default.

**Review by leochr — APPROVED** [review]
@JusteenR Looks good. Thanks.

**Pull request merged**